### PR TITLE
Convert GR103 to the new format

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/CIAC-11525

## Description
In the new GR103, we are converting all warnings in the old version to errors.
 Additionally, removed this validation from the old format to improve performance.(it was un-ignored in the old version).
 
This is blocked by this ticket.
https://jira-dc.paloaltonetworks.com/browse/CIAC-11655

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/GR_validators/GR103_is_using_unknown_content_all_files.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    GenericDefinition,
+    GenericField,
+    GenericModule,
+    GenericType,
+    XSIAMDashboard,
+    XSIAMReport,
+    CaseField,
+    CaseLayout,
+    CaseLayoutRule,
+]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GR_validators/GR103_is_using_unknown_content_all_files.py:80]: We don't need this here, we have the super class that configure that (if it was auto added from the init-script, please fix the init-script)

File: demisto_sdk/commands/validate/validators/GR_validators/GR103_is_using_unknown_content_list_files.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    GenericDefinition,
+    GenericField,
+    GenericModule,
+    GenericType,
+    XSIAMDashboard,
+    XSIAMReport,
+    CaseField,
+    CaseLayout,
+    CaseLayoutRule,
+]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GR_validators/GR103_is_using_unknown_content_list_files.py:80]: We don't need this here, we have the super class that configure that (if it was auto added from the init-script, please fix the init-script)

**Review by YuvHayun — APPROVED** [review]
Nice job, minor comments.

File: .changelog/4522.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Moved GR103 validation to the new validate format. The validation ensures that content items use other content items only if they exist in the repository. Additionally, removed this validation from the old format to improve performance.
```
Comment (YuvHayun) [inline .changelog/4522.yml:2]: split the update about removing from old validate to a bc entry

File: demisto_sdk/commands/validate/tests/GR_validators_test.py [hunk 4]
```diff
@@ -447,3 +453,79 @@ def test_DuplicateContentIdValidatorAllFiles_is_invalid(prepared_graph_repo: Rep
         DuplicateContentIdValidatorAllFiles().obtain_invalid_content_items([])
     )
     assert len(validation_results) == 4
+
+
+@pytest.fixture
+def repo_for_test(graph_repo):
+    # A repository with 3 packs:
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/GR_validators_test.py:460]: let's add more info - what do the 3 packs include?

**Review by YuvHayun — APPROVED** [review]
Looks good!
Minor comments.

File: demisto_sdk/commands/validate/tests/GR_validators_test.py (hunk 4 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/tests/GR_validators_test.py:460]: Thats not my work, obviously...:)

**Pull request merged**